WcaOnRails/app/models/competition.rb

```diff
@@ -962,6 +964,10 @@ def has_event_change_deadline_date?
     start_date.present? && start_date > Date.new(2021, 6, 24)
   end
 
+  def registration_edits_allowed?
+    self.allow_registration_edits && (!has_event_change_deadline_date? || event_change_deadline_date > DateTime.now)
```

Comment:
The downside with that suggestion is that it would prevent organizers from closing registration but leaving edits open.  Let's say that a 50-person competition opened registration, and 100 people signed up in the first hour, so the organizer asked WCAT to close registration.  This would mean that edits are not allowed anymore either, which is unfortunate.  My interpretation was that the reason for having two deadlines was to allow them to be different, and allow event changes after new registrations aren't allowed anymore.

I interpreted all of what you describe as "by design", not by accident :) `self.allow_registration_edits` is there to make sure that no competitions allow edits unintentionally.  Any competition that has it set to true will definitely be after June 24th, 2021, and will have `event_change_deadline_date` set.

I could add validation to ensure that what I just said is true, what do you think?

Reply:
Sorry this fell under my radar. Indeed leaving registration event edits open when the registration is already closed makes a lot of sense, and I probably didn't consider that specific scenario enough when I wrote my previous comments.

No need to add further checks, as I want to avoid "magic dates" in our code as much as possible.